compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -801,15 +804,59 @@ static LogicalResult setMatmulPeelingRootConfig(
   SmallVector<int64_t> vectorReductionTileSizes(numTilingDims, 0);
   std::swap(vectorParallelTileSizes.back(), vectorReductionTileSizes.back());
 
+  // The backend struggles to legalize non-power-of-two scalable vectors,
+  // hence the extra rounding up.
+  for (const auto &[index, size] : llvm::enumerate(vectorParallelTileSizes)) {
+    if (!size)
+      continue;
+    vectorParallelTileSizes[index] =
+        roundUpToPow2(size,
+                      /*predicate=*/inputVecScalableTileFlags[index]);
+  }
```

Comment:
Do we need the adjustment for reduction tile size? How about moving this to l.801 (before setting tile sizes for `vectorParallelTileSizes` and `vectorReductionTileSizes`), and apply it on `clampedVecTileSizes`?

Reply:
> Do we need the adjustment for reduction tile size?

We don't use scalable vectors for the reduction dim, but we might do in do feature. In any case, it will be safer to round up both sets, thanks!

> How about moving this to l.801 (before setting tile sizes for vectorParallelTileSizes and vectorReductionTileSizes), and apply it on clampedVecTileSizes?

I'll move it even further up and keep all the pre-processing for scalable vectors together.